programs/vote/src/vote_state/mod.rs

```diff
@@ -141,40 +141,6 @@ pub fn to<T: WritableAccount>(versioned: &VoteStateVersions, account: &mut T) ->
     VoteState::serialize(versioned, account.data_as_mut_slice()).ok()
 }
 
-// Updates the vote account state with a new VoteState instance.  This is required temporarily during the
-// upgrade of vote account state from V1_14_11 to Current.
-fn set_vote_account_state(
-    vote_account: &mut BorrowedAccount,
-    vote_state: VoteState,
-    feature_set: &FeatureSet,
-) -> Result<(), InstructionError> {
-    // Only if vote_state_add_vote_latency feature is enabled should the new version of vote state be stored
-    if feature_set.is_active(&feature_set::vote_state_add_vote_latency::id()) {
-        // If the account is not large enough to store the vote state, then attempt a realloc to make it large enough.
-        // The realloc can only proceed if the vote account has balance sufficient for rent exemption at the new size.
-        if (vote_account.get_data().len() < VoteStateVersions::vote_state_size_of(true))
-            && (!vote_account
-                .is_rent_exempt_at_data_length(VoteStateVersions::vote_state_size_of(true))
-                || vote_account
-                    .set_data_length(VoteStateVersions::vote_state_size_of(true))
-                    .is_err())
-        {
-            // Account cannot be resized to the size of a vote state as it will not be rent exempt, or failed to be
-            // resized for other reasons.  So store the V1_14_11 version.
-            return vote_account.set_state(&VoteStateVersions::V1_14_11(Box::new(
```

Comment:
this *could* possibly be unsafe if there are still vote states with the old size present when validators start picking up this change.
this seems unlikely since there are no old vote states left and creating new vote states defaults to the new format, however if we wanted to we could additionally feature flag this.

Reply:
There are old vote accounts that never upgraded because they are dead and haven't voted in a long time.

They will be rendered completely unworkable by this change, because every attempt to deserialize their state will fail.  I think this is OK, because they are long dead, but it's something to consider.

Reply:
ah that's a good point. in this case I'll rework this, since that opens us up to an attack vector if they start voting while the cluster is in progress of upgrading to this change.

Reply:
Left the migration code in, will address this with a future flag as removing it is consensus breaking